### PR TITLE
Improve Go cmake, add go/master/c into cmake include path..

### PR DESCRIPTION
**Pull request by helinwang**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -7,18 +7,26 @@ file(GLOB UTILS_PY_FILES . ./paddle/utils/*.py)
...
-             ${UTILS_PY_FILES}
-             ${V2_PY_FILES})
+  ${TRAINER_PY_FILES}
+  ${HELPERS_PY_FILES}
+  ${UTILS_PY_FILES}
+  ${V2_PY_FILES})
+
+add_custom_target(copy_paddle_master)
+if(WITH_GOLANG)
+  add_custom_command(TARGET copy_paddle_master
```
Comment (gangliao) [inline python/CMakeLists.txt]: configure_file(<input> <output> COPYONLY)

File: CMakeLists.txt [hunk 2]
```diff
@@ -130,14 +130,37 @@ if(WITH_GPU)
 endif(WITH_GPU)
 
 add_subdirectory(proto)
-add_subdirectory(paddle)
-add_subdirectory(python)
 
+# "add_subdirectory(paddle)" and "add_subdirectory(python)" should be
+# placed after this block, because they depends on it.
 if(WITH_GOLANG)
```
Comment (gangliao) [inline CMakeLists.txt]: we can put this configuration into `configure.cmake` to keep the root CMAKE FILE as simple as possible.

File: CMakeLists.txt [hunk 3]
```diff
@@ -130,14 +130,37 @@ if(WITH_GPU)
...
-    #TODO (add go/master/c back when fixed)
-    add_subdirectory(go/pserver/cclient)
+  # we need to symlink Paddle directory into GOPATH. If we
+  # don't do it and we have code that depends on Paddle, go
```
Comment (gangliao) [inline CMakeLists.txt]: we can put this configuration into `configure.cmake` to keep the root CMAKE FILE as simple as possible.

File: python/CMakeLists.txt (hunk 1 above)
Comment (helinwang) [inline python/CMakeLists.txt]: Thanks! Done.
Just curios, are there any problem with the old way, or it's just `configure_file` is more ideomatic for copying files.

File: CMakeLists.txt (hunk 3 above)
Comment (helinwang) [inline CMakeLists.txt]: Thanks! Done.

**Pull request merged**